File: src/workshare/tests/fortran/simple.f90
subroutine WS_ASSERT(a,b)
  integer a,b
  if (a .ne. b) then
    write(*,*) 'FAILED !'
    call exit(1)
  endif
  end

  program main
    implicit none
    integer x
    common/x/ x

    integer ( kind = 4 ) i,j,sum,sum2
    x=0 
    !$ws stopsteal
    call WS_ASSERT(x,1)
    !$ws resteal
    call WS_ASSERT(x,2)
    !$ws do reduction(+:sum) private(j) steal_schedule(guided,5) schedule(dynamic,1)
    do i = 1,10
    sum=sum+1 
    end do
    call WS_ASSERT(x,3)

    !$ws critical
    sum2 = sum2 + 1
    !$ws end critical
    call WS_ASSERT(x,4)
    write(*,*) 'SUCCESS !' 
    end

File: src/workshare/tests/simple/simple.c
#include <ws_test.h>

int main(int argc, char ** argv)
{
  int i,j=0,sum=0;
#pragma ws for schedule(guided) reduction(+:sum)
  for(i=0;i<10;i++)
  {
    sum++;
  }

  WS_ASSERT(x,1);

#pragma ws for schedule(dynamic,10) private(j) 
  for(i=0;i<10;i++)
  {
    j++;
    sum++;
  }
  WS_ASSERT(x,2);

#pragma ws stopsteal
  WS_ASSERT(x,3);
#pragma ws resteal
  WS_ASSERT(x,4);
#pragma ws critical
  {
    i++;
  }
  
  WS_ASSERT(x,5);
  WS_SUCCESS();
  return 0;
}
